src/models/edge.ts

```diff
@@ -107,20 +110,30 @@ export interface IEdge<N extends INodeBase, E extends IEdgeBase> {
   getWidth(): number;
   getColor(): Color | string | undefined;
   getLineDashPattern(): number[] | null;
+  setData(data: E): void;
+  setData(callback: (edge: IEdge<N, E>) => E): void;
```

Comment:
Because I think it is more flexible and provides more parameters to play with. While `data: E` in most cases provides only `id`, `start` and `end` (for an edge), `IEdge` provides more parameters that can be used in the callback, like `position` and `style`.

Reply:
Aha ok. Just FYI, `data: E` can be anything that the user wants, at a minimum it has `id`, `start`, `end` (for an edge). Any data the user wants to have on the node/edge will be here (e.g. Lab saves node properties from the database as data).